### PR TITLE
Russell-LA161.Cstorage

### PR DESCRIPTION
**Pull request by russellhz**
#141

### CODE REVIEW

File: R/zchunk_LA161.Cstorage.R [hunk 1]
```diff
@@ -1,66 +1,78 @@
...
+    C_grid_regions <- sort(unique(L161.Csupply_state$grid_region))
+
+    # The construction of supply curves takes place within each grid region.
+    # In each region, there are four cost points assigned, for each quartile of the supply curve
+    region_quartiles <- L161.Csupply_state %>%
+      group_by(grid_region) %>%
+      # Calculate cumulative sum, then filter to the quantiles
+      mutate(Cumul_MtC = cumsum(MtC)) %>%
+      # use "type=1" to indicate that we want to return only exact matches to the data in the table
+      # only take the 2nd to 5th points in the quartiles (25%, 50%, 75%, 100%)
```
Comment (bpbond) [inline R/zchunk_LA161.Cstorage.R]: Nice commenting; thanks.

File: R/zchunk_LA161.Cstorage.R [hunk 2]
```diff
@@ -1,66 +1,78 @@
...
+      filter(Cumul_MtC %in% quantile(Cumul_MtC, type = 1)[2:5]) %>%
+      # From the cumulative totals, return to the marginal quantities associated with each grade
+      mutate(MtC = Cumul_MtC - lag(Cumul_MtC),
+             # this will create grade equal to grade 1 for 25%, grade 2 for 50%, etc.
+             grade = paste("grade",1:n(), sep = " ")) %>%
```
Comment (bpbond) [inline R/zchunk_LA161.Cstorage.R]: `, seq_len(n())` - although I guess in this case it's guaranteed that `n()` ≥ 0

File: inst/extdata/gcam-usa/Dooley_CCS_USA.csv [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# File: Dooley_CCS_USA.csv
```
Comment (gokuliyer) [inline inst/extdata/gcam-usa/Dooley_CCS_USA.csv]: Please include references: 

Dahowski, R., Dooley, J., Davidson, C., Bachu, S. & Gupta, N. Building the Cost Curves for CO2 Storage: North America. IEA Greenhouse Gas R&D Programme, Cheltenham, UK.  (2005).

Wise, M., Dooley, J., Dahowski, R. & Davidson, C. Modeling the impacts of climate policy on the deployment of carbon dioxide capture and geologic storage across electric power regions in the United States. International Journal of Greenhouse Gas Control 1, 261-270 (2007).

Dahowski, R. T., Davidson, C. L., Li, X. C. & Wei, N. A $70/tCO2 greenhouse gas mitigation backstop for China's industrial and electric power sectors: Insights from a comprehensive CCS cost curve. International Journal of Greenhouse Gas Control 11, 73-85, doi:10.1016/j.ijggc.2012.07.024 (2012).

Dahowski, R., Davidson, C. & Dooley, J. Comparing large scale CCS deployment potential in the USA and China: A detailed analysis based on country-specific CO2 transport & storage cost curves. Energy Procedia 4, 2732-2739, doi:DOI: 10.1016/j.egypro.2011.02.175 (2011).

Comment (russellhz) [reply inst/extdata/gcam-usa/Dooley_CCS_USA.csv]: Done!

**Pull request merged**